src/script/conversation/ConversationState.ts

```diff
@@ -32,29 +33,40 @@ import {UserState} from '../user/UserState';
 
 @singleton()
 export class ConversationState {
-  public readonly conversations_cleared: ko.ObservableArray<Conversation>;
-  public readonly sorted_conversations: ko.PureComputed<Conversation[]>;
-  public readonly activeConversation: ko.Observable<Conversation | null>;
+  /**
+   * all the conversations available
+   */
+  public readonly conversations = ko.observableArray<Conversation>([]);
+  /**
+   * current conversation that is being viewed
+   */
+  public readonly activeConversation = ko.observable<Conversation | null>(null);
+  public readonly filteredConversations: ko.PureComputed<Conversation[]>;
+  public readonly archivedConversations: ko.PureComputed<Conversation[]>;
+  public readonly unarchivedConversations: ko.PureComputed<Conversation[]>;
+  public readonly selfConversation: ko.PureComputed<Conversation | undefined>;
   public readonly connectedUsers: ko.PureComputed<User[]>;
-  public readonly conversations_archived: ko.ObservableArray<Conversation>;
-  public readonly conversations_unarchived: ko.ObservableArray<Conversation>;
-  public readonly conversations: ko.ObservableArray<Conversation>;
-  public readonly filtered_conversations: ko.PureComputed<Conversation[]>;
-  public readonly self_conversation: ko.PureComputed<Conversation | undefined>;
+
+  private readonly sortedConversations: ko.PureComputed<Conversation[]>;
 
   constructor(
     private readonly userState = container.resolve(UserState),
     private readonly teamState = container.resolve(TeamState),
   ) {
-    this.activeConversation = ko.observable(null);
-    this.conversations = ko.observableArray([]);
-    this.conversations_archived = ko.observableArray([]);
-    this.conversations_cleared = ko.observableArray([]);
-    this.conversations_unarchived = ko.observableArray([]);
-    this.sorted_conversations = ko.pureComputed(() => this.filtered_conversations().sort(sortGroupsByLastEvent));
-    this.self_conversation = ko.pureComputed(() => this.findConversation(this.userState.self()));
-
-    this.filtered_conversations = ko.pureComputed(() => {
+    this.sortedConversations = ko.pureComputed(() => this.filteredConversations().sort(sortGroupsByLastEvent));
+    this.selfConversation = ko.pureComputed(() =>
+      this.conversations().find(conversation => conversation.type() === CONVERSATION_TYPE.SELF),
```

Comment:
This is the main change. This will prepare for MLS self conversation. 

Previously the selfConversation was found using the userId. Now we use the type of the conversation to know which one is the selfConversation. This is way better semantic and the id of the conversation will probably change with MLS